k8s_example/mwu-encoder/README.md

```diff
@@ -0,0 +1,61 @@
+## 🚀 Deploy MWUEncoder
+
+Visualize the flow:
+
+![flow-stateless](../imgs/flow.svg)
+
+## Steps
+
+**STEP 1**: Start minikube locally:
+
+```python
+minikube start
+# use the Docker daemon from Minikube
+eval $(minikube docker-env)
+```
+
+**STEP 2**: Build docker image:
+
+```
+docker build -f Dockerfile -t mwu_encoder .
+```
+
+**STEP 3**: Deploy to k8s pod:
+
+```
+kubectl apply -f k8s/configmap.yaml 
+kubectl apply -f k8s/deployment.yaml 
+```
+
+It should be noted that we provided a `deployment.yaml` and a `configmap.yaml` in `k8s` folder. `deployment.yaml` specifiy the horizontal scaling capbility with the `replicas` field. Besids, it controls the resource usage and environment variables for the deployment. When applying this deployment, the original entrypoint will be overwritten by the env variables defined in `command`, `args` field and `configmap.yaml`.
```

Comment:
How exactly these replicas behave? Specially for index? do they get the same data all the time?

Reply:
Afaik the compute will be distributed among the replicas so they won't have the exact same data. This is to make the deployment highly available

Reply:
so no relation with replicas in terms of replica of data?

Reply:
hi, this step is more about how to spin up the pods with k8s. We stick to `replicas=1` for the time being.